text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/GsonComponentSerializerImpl.java

```diff
@@ -90,6 +93,27 @@
     return this.serializer().toJson(component);
   }
 
+  @Override
+  public @NonNull Component deserialize(final @NonNull JsonElement input) {
+    final Component component = this.serializer().fromJson(input, Component.class);
+    if(component == null) throw ComponentSerializerImpl.notSureHowToDeserialize(input);
+    return component;
+  }
+
+  @Override
+  public @NonNull Component deserialize(final @NonNull Reader input) {
+    final Component component = this.serializer().fromJson(input, Component.class);
+    if(component == null) throw ComponentSerializerImpl.notSureHowToDeserialize(input);
+    return component;
+  }
+
+  @Override
+  public @NonNull Component deserialize(final @NonNull JsonReader input) {
+    final Component component = this.serializer().fromJson(input, Component.class);
+    if(component == null) throw ComponentSerializerImpl.notSureHowToDeserialize(input);
+    return component;
+  }
+
```

Comment:
Is there really a need for these two separate methods?

Reply:
Not particularly, but `JsonReader` doesn't implement `Reader` so they're two completely different objects.